### ISSUE TITLE
Unable to reuse sg_logger after training

### ISSUE THREAD

**Issue by haritsahm** [Deci-AI/super-gradients#1688]
### 🐛 Describe the bug

Unable to use `sg_logger` aftar training process using `trainer.train(model, ...)`. I want to access the wandb metadata (run_id, experiment_name, project_name, etc.) to upload additional file into the same experiment log after the training process.

```
from super_gradients import Trainer
from super_gradients.common.sg_loggers import BaseSGLogger, WandBSGLogger

# create a trainer object, look the declaration for more parameters
trainer = Trainer("experiment_name")

train_params = { ... # training parameters
                "sg_logger": "wandb_sg_logger", # Weights&Biases Logger, see class WandBSGLogger for details
                "sg_logger_params": # paramenters that will be passes to __init__ of the logger 
                  {
                    "project_name": "project_name", # W&B project name
                    "save_checkpoints_remote": True
                    "save_tensorboard_remote": True
                    "save_logs_remote": True
                  } 
               }

trainer.train(model=model, training_params=train_params, train_loader=train_dataloader, valid_loader=val_dataloader)

# Validate sg_logger
if trainer.sg_logger is not None:
    if isinstance(trainer.sg_logger, BaseSGLogger):
        print("Found BaseSGLogger type for sg_logger")
    elif isinstance(trainer.sg_logger, WandBSGLogger):
        print("Found WandBSGLogger type for sg_logger")
    else:
        print(f"Found other type for sg_logger: {type(trainer.sg_logger)}")
else:
    print("trainer.sg_logger is not valid")
```

Output
```
trainer.sg_logger is not valid
```

### Versions

PyTorch version: 2.1.1+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.23.3
Libc version: glibc-2.31

Python version: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:10)  [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-89-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1060 6GB
Nvidia driver version: 545.23.08
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.5.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.5.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.5.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.5.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.5.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.5.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.5.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      43 bits physical, 48 bits virtual
CPU(s):                             12
On-line CPU(s) list:                0-11
Thread(s) per core:                 2
Core(s) per socket:                 6
Socket(s):                          1
NUMA node(s):                       1
Vendor ID:                          AuthenticAMD
CPU family:                         23
Model:                              113
Model name:                         AMD Ryzen 5 3600 6-Core Processor
Stepping:                           0
Frequency boost:                    enabled
CPU MHz:                            2200.000
CPU max MHz:                        3600.0000
CPU min MHz:                        2200.0000
BogoMIPS:                           7200.33
Virtualization:                     AMD-V
L1d cache:                          192 KiB
L1i cache:                          192 KiB
L2 cache:                           3 MiB
L3 cache:                           32 MiB
NUMA node0 CPU(s):                  0-11
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Mitigation; untrained return thunk; SMT enabled with STIBP protection
Vulnerability Spec rstack overflow: Mitigation; safe RET
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca sme sev sev_es

Versions of relevant libraries:
[pip3] flake8==6.1.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.22.4
[pip3] onnx==1.13.0
[pip3] onnx-simplifier==0.4.35
[pip3] onnxruntime==1.13.1
[pip3] pytorch-lightning==2.1.0
[pip3] pytorch-quantization==2.1.2
[pip3] torch==2.1.1
[pip3] torch-tensorrt==1.2.0a0
[pip3] torchinfo==1.8.0
[pip3] torchmetrics==0.8.0
[pip3] torchtext==0.11.0a0
[pip3] torchvision==0.16.1
[pip3] triton==2.1.0
[conda] mkl                       2020.4             h726a3e6_304    conda-forge
[conda] mkl-include               2020.4             h726a3e6_304    conda-forge
[conda] numpy                     1.22.4           py38h99721a1_0    conda-forge
[conda] pytorch-lightning         2.1.0                    pypi_0    pypi
[conda] pytorch-quantization      2.1.2                    pypi_0    pypi
[conda] torch                     2.1.1                    pypi_0    pypi
[conda] torch-tensorrt            1.2.0a0                  pypi_0    pypi
[conda] torchinfo                 1.8.0                    pypi_0    pypi
[conda] torchmetrics              0.8.0                    pypi_0    pypi
[conda] torchtext                 0.11.0a0                 pypi_0    pypi
[conda] torchvision               0.16.1                   pypi_0    pypi
[conda] triton                    2.1.0                    pypi_0    pypi

**Comment by Louis-Dupont** [comment]
Hi @haritsahm ,

I did not manage to reproduce, here is the code I just ran to test it:
``` python
from super_gradients import Trainer
from super_gradients.training.metrics.classification_metrics import Accuracy, Top5
from torch.optim import ASGD
from torch.optim.lr_scheduler import MultiStepLR, ReduceLROnPlateau
from torch.nn import CrossEntropyLoss
from super_gradients.training.utils.callbacks import Phase, LRSchedulerCallback
from torchvision.datasets import CIFAR10
from torchvision.transforms import ToTensor
from torchvision.models import resnet18
from torch.utils.data import DataLoader

# Define any torch DataLoaders, need at least train & valid loaders
train_dataset = CIFAR10(root="data/", download=True, train=True, transform=ToTensor())
valid_dataset = CIFAR10(root="data/", download=True, train=False, transform=ToTensor())

train_loader = DataLoader(train_dataset, shuffle=True, batch_size=16)
valid_loader = DataLoader(valid_dataset, batch_size=32)

# Define any network of type nn.Module
net = resnet18(num_classes=len(train_dataset.classes))

# Define any optimizer of type torch.optim.Optimizer (and schedulers)
lr = 2.5e-4
optimizer = ASGD(net.parameters(), lr=lr, weight_decay=0.0001)

rop_lr_scheduler = ReduceLROnPlateau(optimizer, mode="max", patience=10, verbose=True)
step_lr_scheduler = MultiStepLR(optimizer, milestones=[0, 150, 200], gamma=0.1)

# Define any loss function of type torch.nn.modules.loss._Loss
loss_fn = CrossEntropyLoss()

# Define phase callbacks
phase_callbacks = [
    LRSchedulerCallback(scheduler=rop_lr_scheduler, phase=Phase.VALIDATION_EPOCH_END, metric_name="Accuracy"),
    LRSchedulerCallback(scheduler=step_lr_scheduler, phase=Phase.TRAIN_EPOCH_END),
]

# Bring everything together with Trainer and start training
trainer = Trainer("Cifar10_external_objects_example")

train_params = {
    "max_epochs": 1,
    "phase_callbacks": phase_callbacks,
    "loss": loss_fn,
    "criterion_params": {},
    "optimizer": optimizer,
    "train_metrics_list": [Accuracy(), Top5()],
    "valid_metrics_list": [Accuracy(), Top5()],
    "metric_to_watch": "Accuracy",
    "greater_metric_to_watch_is_better": True,
    "lr_scheduler_step_type": "epoch",
}

train_params.update(
    {
        "sg_logger": "wandb_sg_logger",  # Weights&Biases Logger, see class WandBSGLogger for details
        "sg_logger_params": {  # paramenters that will be passes to __init__ of the logger
            "project_name": "project_name",  # W&B project name
            "save_checkpoints_remote": True,
            "save_tensorboard_remote": True,
            "save_logs_remote": True,
        },
    }
)


trainer.train(model=net, training_params=train_params, train_loader=valid_loader, valid_loader=valid_loader)

print(trainer.sg_logger)
```

Could you try and see if you still get None ?
Also, are you sure `trainer.train` was called in your script ? I'm asking because the logger is instantiated when `trainer.train()` is called so if you commented it out to debug faster it may explain why `trainer.sg_logger is None`.

Let me know if the code works. Otherwise, it would help if you can provide a minimalistic script that we could copy/paste to reproduce

**Comment by BloodAxe** [comment]
Closing as stale

**Comment by haritsahm** [comment]
Yes there was an error on my side, not from sg
